src/compute/src/sink/persist_sink.rs

```diff
@@ -0,0 +1,195 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::any::Any;
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::ops::DerefMut;
+use std::rc::Rc;
+use std::time::Duration;
+
+use differential_dataflow::{Collection, Hashable};
+use timely::dataflow::channels::pact::Exchange;
+use timely::dataflow::operators::generic::builder_rc::OperatorBuilder;
+use timely::dataflow::Scope;
+use timely::progress::Antichain;
+use timely::progress::Timestamp as TimelyTimestamp;
+
+use mz_dataflow_types::sinks::{PersistSinkConnector, SinkDesc};
+use mz_persist_client::Location;
+use mz_repr::{Diff, GlobalId, Row, Timestamp};
+use mz_timely_util::operators_async_ext::OperatorBuilderExt;
+
+use crate::render::sinks::SinkRender;
+
+impl<G> SinkRender<G> for PersistSinkConnector
+where
+    G: Scope<Timestamp = Timestamp>,
+{
+    fn uses_keys(&self) -> bool {
+        false
+    }
+
+    fn get_key_indices(&self) -> Option<&[usize]> {
+        None
+    }
+
+    fn get_relation_key_indices(&self) -> Option<&[usize]> {
+        None
+    }
+
+    fn render_continuous_sink(
+        &self,
+        compute_state: &mut crate::compute_state::ComputeState,
+        _sink: &SinkDesc,
+        sink_id: GlobalId,
+        sinked_collection: Collection<G, (Option<Row>, Option<Row>), Diff>,
+    ) -> Option<Rc<dyn Any>>
+    where
+        G: Scope<Timestamp = Timestamp>,
+    {
+        let scope = sinked_collection.scope();
+        let operator_name = format!("persist_sink({})", self.shard_id);
+        let mut persist_op = OperatorBuilder::new(operator_name, scope.clone());
+
+        // We want exactly one worker (in the cluster) to send all the data to persist. It's fine
+        // if other workers from replicated clusters write the same data, though. In the real
+        // implementation, we would use a storage client that transparently handles writing to
+        // multiple shards. One shard would then only be written to by one worker but we get
+        // parallelism from the sharding.
```

Comment:
I'd maybe toss an explicit `// TODO` here, as I think we'll want to sort this out somewhat promptly. Like, no reason afaict that we shouldn't just be writing to one persist shard per worker.

Reply:
Will put in a `TODO`